### ISSUE TITLE
Registration process is not cancelled if the user registration failed

### ISSUE THREAD

**Issue by pixeljunkie** [chameleon-system/chameleon-system#681]
**Describe the bug**
If the user registration failed, the extranet component and the user data step of the step wizard module do not handle this and go on to the next process like it was a success.
This may rarely happen in the wild if you are working with the core modules without custom extensions.
But if you connect the registration to a third party user database this may occur (connection error, username already taken, any other API error).

**Affected version(s)**
7.0.x and earlier

**To Reproduce**
There is no way to reproduce this without extending the code, but the errors in the code are impossible to be unseen.
1. temporary change TdbDataExtranetUser::Register() and prevent saving a record in the database.
2. return always false in this method
3. If you register now during the order process or via registration page you should not succeed... but you do, kind of.

**Expected behavior**
A failed registration needs to send the user back to the form he came from and show an registration error.

**Technical details**
We already fixed both issues in a customer project.

The Hotfix:

TShopStepUserDataV2EndPoint::UpdateUser()

```php
// added:
 if (false === $oUser->Register()) {
   throw new UserRegistrationFailedException('User registration failed');
}

public function ProcessStep()
    {
        try {
            return parent::ProcessStep();
        } catch (UserRegistrationFailedException $e) {
            return false;
        }
    }
```

MTExtranetCore:.Register()

```php
$success = $oUser->Register();
                if($success) {
                    $this->UpdateUserAddress(null, null, true);
...
```

**Comment by UlrichKu** [comment]
This now breaks guest "login" for them the "update user" step is now always false.

**Comment by esono-anja** [comment]
I've checked the normal registration process to make sure that this is still ok. But it's not. The following behaviour occurs:
1. Complete the registration form
2. Hit the button "Register"
3. See error: the form remains, additionally the field date of birth is filled in with 0000-00-00.
=> expectation: The message "registration successful" appears.

**Comment by esono-anja** [comment]
fixed for 7.1
